### ISSUE TITLE
babel-node, .babelrc, yarn run and cli arguments do not fit together

### ISSUE THREAD

**Issue by terion-name** [babel/babel#7932]
## Bug Report

After update of `.bablrc` behaviour in beta.46 `run` commands started to fail. I have in `package.json`:

```js
"scripts": {
    "build": "./node_modules/.bin/babel-node src/build.js",
  }
```

If i run `yarn run build` it fails to allocate `.babelrc` in project root (fails with `SyntaxError: Unexpected token import` at script imports)

If I run `./node_modules/.bin/babel-node src/build.js` directly — it works fine.

Something like `yarn --cwd=$PWD run build` does not help.

Ok, if I remove `.babelrc` and add `package.json` following:
```js
"babel": {
    "presets": ["@babel/es2017", "@babel/es2016", "@babel/es2015", "@babel/flow"],
    "plugins": ["@babel/proposal-object-rest-spread"]
  }
```
Then `yarn run build` starts to work. 

BUT!

If I try to pass arguments for the script:
```
yarn run build -- -o foo
```

It fails again with the same error.

The same is with direct runs: `./node_modules/.bin/babel-node ./src/build.js` works, `./node_modules/.bin/babel-node ./src/build.js -o foo` fails

All the same with usage of `babel.config.js`

## UPD
using this:
```js
require('@babel/register');
require('@babel/polyfill');
require('./_build.js');
```
everything works

**Comment by sant123** [comment]
Hi, I'm using `babel www/js/main.jsx -o www/js/main.js` but it seems is not reading .babelrc. This is my project structure:

![image](https://user-images.githubusercontent.com/7959437/41556711-2c6740d2-7301-11e8-9a22-8ac76d4a2d6e.png)

My workaroung is to explicitely set --config-file like this:

`babel www/js/main.jsx --config-file ./.babelrc -o www/js/main.js`

¿Is this the intented behavior?

If it does `--no-babelrc` will turn useless. 

Thank you!

**Comment by sant123** [comment]
I forgot to mention I'm using babel 7 and now we can use `babel.config.js`. I renamed .babelrc to it and now I have:

``` js
module.exports = {
  presets: ["@babel/preset-react"]
};
```

My command `babel www/js/main.jsx -o www/js/main.js` now works without --config-file.

**Comment by loganfsmyth** [comment]
@terion-name Sorry this got missed. If you're still having issues, it's a little hard to follow the structure that you'e describing. If you can make an example repo, or at least a clearer description of the directory structure and config file placement, I can try to help.

**Comment by nicolo-ribaudo** [comment]
Closing since it's not clear what the bug is. As mentioned in https://github.com/babel/babel/issues/7932#issuecomment-398173001, this seems to work in Babel 7.
